### PR TITLE
[Dist] defer to load node/edge feats

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Description

CPU RAM is tight sometimes, let's defer loading of node/edge feats to lower the peak memory consumption.
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/distributed/partition.py [hunk 1]
```diff
@@ -61,6 +61,8 @@ def load_partition(part_config, part_id):
         The path of the partition config file.
     part_id : int
         The partition ID.
+    load_feats : bool, optional
+        Whether to load node/edge feats. Default: True.
```
Comment (jermainewang) [inline python/dgl/distributed/partition.py:65]:
```suggestion
        Whether to load node/edge feats. If False, the returned node/edge feature dictionaries 
        will be empty. Default: True.
```

File: python/dgl/distributed/partition.py [hunk 2]
```diff
@@ -134,8 +116,61 @@ def load_partition(part_config, part_id):
...
+        Edge features.
+    '''
+    config_path = os.path.dirname(part_config)
+    relative_to_config = lambda path: os.path.join(config_path, path)
+
+    with open(part_config) as conf_f:
+        part_metadata = json.load(conf_f)
+    assert 'part-{}'.format(part_id) in part_metadata, "part-{} does not exist".format(part_id)
+    part_files = part_metadata['part-{}'.format(part_id)]
+    assert 'node_feats' in part_files, "the partition does not contain node features."
```
Comment (jermainewang) [inline python/dgl/distributed/partition.py:151]: "Invalid partition file format: missing node features."

File: python/dgl/distributed/partition.py [hunk 3]
```diff
@@ -134,8 +116,61 @@ def load_partition(part_config, part_id):
...
+    assert 'edge_feats' in part_files, "the partition does not contain edge feature."
```
Comment (jermainewang) [inline python/dgl/distributed/partition.py:152]: "Invalid partition file format: missing edge features."

File: tests/distributed/test_partition.py [hunk 4]
```diff
@@ -180,7 +180,12 @@ def check_hetero_partition(hg, part_method, num_parts=4, num_trainers_per_machin
     shuffled_labels = []
     shuffled_elabels = []
     for i in range(num_parts):
-        part_g, node_feats, edge_feats, gpb, _, ntypes, etypes = load_partition('/tmp/partition/test.json', i)
+        part_g, node_feats, edge_feats, gpb, _, ntypes, etypes = load_partition(
+            '/tmp/partition/test.json', i, load_feats=load_feats)
+        if not load_feats:
+            assert not node_feats
+            assert not edge_feats
+            node_feats, edge_feats = load_partition_feats('/tmp/partition/test.json', i)
```
Comment (jermainewang) [inline tests/distributed/test_partition.py:188]: How about
```python
        if not load_feats:
            assert not node_feats
            assert not edge_feats
        else:
            node_feats1, edge_feats1 = load_partition_feats('/tmp/partition/test.json', i)
            # ... verify node/edge_feats1 and node/edge_feats are the same
```

File: tests/distributed/test_partition.py [hunk 5]
```diff
@@ -252,7 +257,12 @@ def check_partition(g, part_method, reshuffle, num_parts=4, num_trainers_per_mac
     shuffled_labels = []
     shuffled_edata = []
     for i in range(num_parts):
-        part_g, node_feats, edge_feats, gpb, _, ntypes, etypes = load_partition('/tmp/partition/test.json', i)
+        part_g, node_feats, edge_feats, gpb, _, ntypes, etypes = load_partition(
+            '/tmp/partition/test.json', i, load_feats=load_feats)
+        if not load_feats:
+            assert not node_feats
+            assert not edge_feats
+            node_feats, edge_feats = load_partition_feats('/tmp/partition/test.json', i)
```
Comment (jermainewang) [inline tests/distributed/test_partition.py:265]: same comment as above

File: tests/distributed/test_partition.py (hunk 4 above)
Comment (Rhett-Ying) [reply tests/distributed/test_partition.py:188]: I'd like to test `load_partition(..., load_feats=False)` as well. So I think current one is more suitable and the correctness of `node/edge_feats` is verified in below.

File: tests/distributed/test_partition.py (hunk 5 above)
Comment (Rhett-Ying) [reply tests/distributed/test_partition.py:265]: same reply as above...

**Pull request merged**